mscore/plugin/api/elements.h

```diff
@@ -636,6 +638,72 @@ class Measure : public Element {
       /// \endcond
       };
 
+//---------------------------------------------------------
+//   BSymbol
+//    BSymbol wrapper
+//---------------------------------------------------------
+
+class BSymbol : public Element {
+      /// \since MuseScore 3.3.1
+      Q_OBJECT
+      Q_PROPERTY(QQmlListProperty<Ms::PluginAPI::Element>  elements          READ elements)
+
+   public:
+      /// \cond MS_INTERNAL
+      BSymbol(Ms::BSymbol* o = nullptr, Ownership own = Ownership::PLUGIN)
+         : Element(o, own) {}
+
+      Ms::BSymbol* bsymbol() { return toBSymbol(e); }
+      const Ms::BSymbol* bsymbol() const { return toBSymbol(e); }
+
+      QQmlListProperty<Element> elements() { return wrapContainerProperty<Element>(this, bsymbol()->leafs());   }
+      /// \endcond
+      };
+
+//---------------------------------------------------------
+//   Image
+//    Image wrapper
+//---------------------------------------------------------
+
+class Image : public BSymbol {
+      /// \since MuseScore 3.3.1
+      Q_OBJECT
+      Q_PROPERTY(QString id   READ id)
+
+   public:
+      /// \cond MS_INTERNAL
+      Image(Ms::Image* o = nullptr, Ownership own = Ownership::PLUGIN)
+         : BSymbol(o, own) {}
+
+      Ms::Image* image()             { return toImage(e); }
+      const Ms::Image* image() const { return toImage(e); }
+
+      QString id() const             { return image()->id(); }
+      /// \endcond
+      };
+
+//---------------------------------------------------------
+//   Symbol
+//    Symbol wrapper
+//---------------------------------------------------------
+
+class Symbol : public BSymbol {
+      /// \since MuseScore 3.3.1
+      Q_OBJECT
+      Q_PROPERTY(QString id  READ id)
+
+   public:
+      /// \cond MS_INTERNAL
+      Symbol(Ms::Symbol* o = nullptr, Ownership own = Ownership::PLUGIN)
+         : BSymbol(o, own) {}
+
+      Ms::Symbol* symbol()             { return toSymbol(e); }
+      const Ms::Symbol* symbol() const { return toSymbol(e); }
+
+      QString id() const               { return symbol()->symName(); }
```

Comment:
We already have a property `symbol` in `Element` objects which gives an access to a property with `Pid::SYMBOL`. Wouldn't it be better to just use it and expose `SymId` enumeration to plugins instead?

Reply:
What you propose was similar to my first implementation, but it proved clumsy from the plug-in point of view. Images and symbols (font glyphs) go together in a number of contexts, for instance unsupported ornaments or other specialised / niche score markings: the font glyph when one exists / a custom image when it does not.

The only way to identify an image is through its file name, i.e. a string; if symbols are identified through a number (the `SymId` enum) then the plug-in needs to duplicate most of its code to manage images on one side and font glyphs on the other side.

In this way, images and font glyphs have the same interface and can be managed with the same (plug-in) code.

Reply:
Could you give an example of such plugin code? I don't see currently how having two properties named differently might lead to duplicating most of plugin's code.

I would prefer to avoid adding a property like `id` as it:
- duplicates already existing properties (at least for Symbol)
- has a very broad name which makes it unclear what exactly does this property stand for.